File: easy/roller_coaster.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    chomp($line);
    my $up = 1;
    my @r  = map {
        if ( $_ =~ /^[[:alpha:]]/ )
        {
            $up = !$up;
            if ($up) {
                lc $_;
            }
            else {
                uc $_;
            }
        }
        else {
            $_;
        }
    } split( //, $line );
    printf( "%s\n", join( '', @r ) );
}
close(INFILE);

File: easy/roller_coaster.rb
File.open(ARGV[0]).each_line do |line|
  up = true
  line.chomp.each_char do |i|
    if i =~ /[[:alpha:]]/
      print up ? i.upcase : i.downcase
      up = !up
    else
      print i
    end
  end
  puts
end
